### PR TITLE
Added actions for GVA buttons

### PR DESCRIPTION
**Pull request by EgorovEI**
Added unit tests for GVA button actions

MOB-2378

### CODE REVIEW

**Comment by EgorovEI** [comment]
http(s)-based url opening should discussed in universal link perspective, so it might be changed.

File: GliaWidgets/Sources/ViewModel/Chat/ChatViewModel+GVA.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+private extension String {
+    static let gvaOptionUrlTargetModal = "modal"
+    static let gvaOptionUrlTargetSelf = "self"
+}
+
+extension ChatViewModel {
+    func gvaOptionAction(for option: GvaOption) -> Cmd {
+        // If option contains `url`, then it's URL Button,
+        // otherwise it's Postback Button and option should be sent
+        // to the server as `SingleChoiceOption`
```
Comment (gersonnoboa) [inline GliaWidgets/Sources/ViewModel/Chat/ChatViewModel+GVA.swift:13]: This is good information

**Review by gersonnoboa — COMMENTED** [review]
Is it fine that both actions just return if there is a failure, instead of giving some kind of error?

**Comment by EgorovEI** [comment]
> Is it fine that both actions just return if there is a failure, instead of giving some kind of error?

Sending SingleChoiseOption failure is handled by presenting an alert.

There is only one case with opening urls is not handled, it's deeplinks without proper urlTarget value. In this case nothing happens. Did you mean this one?

**Comment by gersonnoboa** [comment]
@EgorovEI yeah the return in the default and also for the guards. Maybe the guards are not important, but if the return in the default is correct, then all is well.

**Review by gersonnoboa — APPROVED** [review]

**Comment by EgorovEI** [comment]
> yeah the return in the default and also for the guards. Maybe the guards are not important, but if the return in the default is correct, then all is well.

Thank you for raising this question. For now we covered all cases we need to handle. Some things that might be improved we will discuss later.

File: GliaWidgets/Sources/ViewModel/Chat/ChatViewModel+GVA.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        if let urlString = option.url,
+           let url = URL(string: urlString) {
+            return urlButtonAction(url: url, urlTarget: option.urlTarget)
+        } else {
```
Comment (rasmustautsglia) [inline GliaWidgets/Sources/ViewModel/Chat/ChatViewModel+GVA.swift:17]: I would add another case for a broadcast event that logs an error. So if a broadcast event is present regardless of other values, then error should be logged but not presented

**Review by igorkravchenko — APPROVED** [review]

**Review by rasmustautsglia — APPROVED** [review]

File: GliaWidgets/Sources/ViewModel/Chat/ChatViewModel+GVA.swift (hunk 2 above)
Comment (EgorovEI) [reply GliaWidgets/Sources/ViewModel/Chat/ChatViewModel+GVA.swift:17]: Good point! 
I will add presenting an alert indicating that broadcast events are not supported as mentioned in project plan

**Comment by EgorovEI** [comment]
!squash

**Pull request merged**